src/pytorch_lightning/strategies/dp.py

```diff
@@ -125,8 +125,8 @@ def barrier(self, *args: Any, **kwargs: Any) -> None:
     def broadcast(self, obj: TBroadcast, src: int = 0) -> TBroadcast:
         return obj
 
-    def reduce_boolean_decision(self, decision: bool) -> bool:
-        return decision
+    def reduce_boolean_decision(self, decision: Union[bool, Tensor]) -> bool:
```

Comment:
I disagree with the change. Let's instead convert to bool where necessary when calling the method. I don't see a reason why we would want to allow a tensor.

Reply:
In the current `ModelCheckpoint` implementation, `reduce_boolean_decision` is called with a Tensor input: https://github.com/Lightning-AI/lightning/blob/35ec79ad1f75790767a6857637c9bbe052e47b58/src/pytorch_lightning/callbacks/model_checkpoint.py#L496

The current implementation in master supports this, but it's inefficient. It converts the tensor to int and then back to a tensor: https://github.com/Lightning-AI/lightning/blob/35ec79ad1f75790767a6857637c9bbe052e47b58/src/pytorch_lightning/strategies/parallel.py#L116

So master already allows a Tensor, the changes in this PR just fix the type and also makes the implementation a bit better by not creating a new tensor type.

@awaelchli do you still want to manually convert to `bool` in `ModelCheckpoint` before calling `reduce_boolean_decision`?

Reply:
Yes I prefer to keep it to bool. Otherwise one has to deal with non-scalar tensors and that just makes it unnecessarily complicated.

Reply:
Done